### PR TITLE
Problem: need a common cxxtools JSON serializer/parser simple interface

### PR DESCRIPTION
**Pull request by jeffdegott**
Solution: do it
* cxxtools JSON serializer/parser simple interface
Signed-off-by: Degott, Francois Regis <FrancoisRegisDegott@eaton.com>

### CODE REVIEW

File: include/fty_common_json.h [hunk 1]
```diff
@@ -82,6 +89,49 @@ namespace JSON {
...
+    //
+    // cxxtools SerializationInfo interface
+    //
+
+    /**
+     * \brief Read/set a SerializationInfo object from a JSON file.
+     * \param[in]   path_name - the path to the JSON file
+     * \param[out]  si - cxxtools::SerializationInfo object
+     * \return  void
+     * \throw generic exceptions
```
Comment (eric3873) [inline include/fty_common_json.h]: Developing the extensions name could be appreciated

File: project.xml [hunk 2]
```diff
@@ -42,6 +42,13 @@
     <version major = "1" minor = "0" patch = "0" />
     <abi current = "1" revision = "0" age = "0" />
 
+    <use project = "cxxtools"
+        test = "cxxtools::Utf8Codec::Utf8Codec"
+        header = "cxxtools/allocator.h"
+        repository = "https://github.com/42ity/cxxtools.git"
+        release = "2.2-FTY-master"
```
Comment (eric3873) [inline project.xml]: Do you check the version 3.0 ?

File: src/fty_common_json.cc [hunk 3]
```diff
@@ -96,6 +95,92 @@ std::string readString (const std::string &line, size_t &start_pos, size_t &end_
...
+// cxxtools SerializationInfo interface
+//
+
+// write SI to JSON string
+std::string writeToString (cxxtools::SerializationInfo& si)
+{
+    try {
+        std::ostringstream output;
+        cxxtools::JsonSerializer serializer;
+        serializer.beautify(true);
```
Comment (eric3873) [inline src/fty_common_json.cc]: An option for beautify ?

File: src/fty_common_json.cc [hunk 4]
```diff
@@ -153,5 +240,162 @@ fty_common_json_test (bool verbose)
...
+        try {
+            buffer = "{}";
+            JSON::readFromString(buffer, si);
+        }
+        catch (const std::exception& e) {
+            failed = true;
+        }
+        assert(!failed);
+
+        // empty JSON buffer
```
Comment (eric3873) [inline src/fty_common_json.cc]: Typo error

File: src/fty_common_json.cc [hunk 5]
```diff
@@ -153,5 +240,162 @@ fty_common_json_test (bool verbose)
...
+        failed = false;
+        try {
+            buffer = "{\"member1\": \"value1\", \"member2\": 42, \"array\": [{\"x\": 12}, {\"x\": 13}]}";
+            JSON::readFromString(buffer, si);
+        }
+        catch (const std::exception& e) {
+            failed = true;
+        }
+        assert(!failed);
+    }
```
Comment (eric3873) [inline src/fty_common_json.cc]: A test with bad json format ?

File: include/fty_common_json.h [hunk 6]
```diff
@@ -82,6 +89,51 @@ namespace JSON {
...
+
+    /**
+     * \brief Write a SerializationInfo object into a JSON string.
+     * \param[int]  si - cxxtools::SerializationInfo object
+     * \param[in]  beautify - beautify'er
+     * \return  the result string
+     * \throw generic exceptions
+     */
+    std::string writeToString (cxxtools::SerializationInfo& si, bool beautify = true);
+
```
Comment (boricj) [inline include/fty_common_json.h]: Add reading/writing from C++ streams.

**Pull request merged**